descqa/SizeStellarMassLuminosity.py

```diff
@@ -90,12 +97,30 @@ def run_on_single_catalog(self, catalog_instance, catalog_name, output_dir):
         catalog_data = catalog_instance.get_quantities(list(colnames.values()), filters=filters)
         catalog_data = {k: catalog_data[v] for k, v in colnames.items()}
 
-        fig, axes = plt.subplots(2, 3, figsize=(9, 6), sharex=True, sharey=True)
+
+        fig, axes = plt.subplots(self.fig_subplot_row, self.fig_subplot_col, figsize=(self.fig_subplot_col*4, self.fig_subplot_row*4), sharex=True, sharey=True)
+
+        if catalog_name.lower()[:5] == 'proto': 
```

Comment:
Validation tests should *not* adjust itself by checking the catalog name. If these adjustments are needed, use the *content* of the catalog as conditions. For example, the test can check if there are enough galaxies in a redshift bin to decide whether or not to plot that bin, but not by checking the catalog name.

Reply:
I am not quite sure what you meant by this. I have a variable *catalog_name* in that function and I was considering a few characters of that variable. I am not able to understand this

Reply:
Yes, `catalog_name` is provided but it should only be used for printing labels, etc. The logic of the test should NOT differ for different catalogs. That is because, for example, one that another catalog that is named protoDC3 may have totally different properties than protoDC2. Hence, we should design the test so that it does not need to depend on `catalog_name`.